server/api/src/rsGetHostForGet.cpp

```diff
@@ -17,8 +17,73 @@
 #include "irods/rsGetHostForGet.hpp"
 #include "irods/irods_resource_backport.hpp"
 #include "irods/irods_resource_redirect.hpp"
+#include "irods/irods_at_scope_exit.hpp"
 
+#include <nlohmann/json.hpp>
+
+#include <set>
 #include <cstring>
+#include <optional>
+
+int rs_get_resource_info_for_operation(
+    rsComm_t*     rsComm,
+    dataObjInp_t* dataObjInp,
+    char **       outInfo )
+{
+    rodsServerHost_t *rodsServerHost;
+    const int remoteFlag = getAndConnRemoteZone(rsComm, dataObjInp, &rodsServerHost, REMOTE_OPEN);
+    if (remoteFlag < 0) {
+        return remoteFlag;
+    }
+    else if (REMOTE_HOST == remoteFlag) {
+        const int status = rc_get_resource_info_for_operation(rodsServerHost->conn, dataObjInp, outInfo);
+        if (status < 0) {
+            return status;
+        }
```

Comment:
We could avoid the `else if` (i.e. just use `if`) and the `else` block if we did this (or something like it):
```suggestion
        return rc_get_resource_info_for_operation(rodsServerHost->conn, dataObjInp, outInfo);
```

Reply:
I dunno.  The existing code, copied from rsGetHostFor(Put,Get).cpp, appears to want to flatten any status response it gets that is greater than zero, to 0 via the return 0 that comes at function's end.

I'll the conservative viewpoint of leaving as-is, I think, for now, as maybe that was done for reasons.
But I'm willing to listen to reason.

Reply:
done